pkg/cfn/builder/vpc.go

```diff
@@ -163,11 +163,13 @@ func (c *ClusterResourceSet) addOutputsForVPC() {
 		c.spec.VPC.ID = v
 		return nil
 	})
-	c.rs.defineOutputWithoutCollector(outputs.ClusterFeatureNATMode, c.spec.VPC.NAT.Gateway, false)
-	if refs, ok := c.subnets[api.SubnetTopologyPrivate]; ok {
-		c.rs.defineJoinedOutput(outputs.ClusterSubnetsPrivate, refs, true, func(v string) error {
-			return vpc.ImportSubnetsFromList(c.provider, c.spec, api.SubnetTopologyPrivate, strings.Split(v, ","))
-		})
+	if c.spec.VPC.NAT != nil {
+		c.rs.defineOutputWithoutCollector(outputs.ClusterFeatureNATMode, c.spec.VPC.NAT.Gateway, false)
+		if refs, ok := c.subnets[api.SubnetTopologyPrivate]; ok {
```

Comment:
just want to confim, so we will `**only**` export private subnets if `NAT` is not empty ?

Reply:
Good catch, fixed. Thank you, @sayboras!

Reply:
you are welcome :100: